src/encoded/commands/deploy.py

```diff
@@ -373,26 +395,58 @@ def _get_run_args(main_args, instances_tag_data):
     return run_args
 
 
+def _get_instance_output(instances_tag_data, is_cluster_master=False):
+    suffix = '-dm' if is_cluster_master else ''
+    hostname = '{}.{}.encodedcc.org'.format(
+        instances_tag_data['id'],
+        instances_tag_data['domain'],
+    )
+    return [
+        'Host %s%s.*' % (instances_tag_data['short_name'], suffix),
+        '  Hostname %s' % hostname,
+        '  # https://%s.demo.encodedcc.org' % instances_tag_data['name'],
+        '  # ssh %s' % hostname,
+        '  # %s' % instances_tag_data['id'],
+    ]
+
+
 def _wait_and_tag_instances(main_args, run_args, instances_tag_data, instances, cluster_master=False):
     tmp_name = instances_tag_data['name']
-    domain = 'production' if main_args.profile_name == 'production' else 'instance'
-    for i, instance in enumerate(instances):
-        if main_args.elasticsearch == 'yes' and run_args['count'] > 1:
-            if cluster_master and run_args['master_user_data']:
-                print('Creating Elasticsearch Master Node for cluster')
-                # Hack: current tmp_name was the last data cluster, so remove '4'
-                instances_tag_data['name'] = "{}{}".format(tmp_name[0:-1], 'master')
-            else:
-                print('Creating Elasticsearch cluster')
-                instances_tag_data['name'] = "{}{}".format(tmp_name, i)
+    instances_tag_data['domain'] = 'production' if main_args.profile_name == 'production' else 'instance'
+    output_list = ['']
+    is_cluster_master = False
+    is_cluster = False
+    if main_args.elasticsearch == 'yes' and run_args['count'] > 1:
+        if cluster_master and run_args['master_user_data']:
+            is_cluster_master = True
+            output_list.append('Creating Elasticsearch Master Node for cluster')
         else:
-            instances_tag_data['name'] = tmp_name
+            is_cluster = True
+            output_list.append('Creating Elasticsearch cluster')
+    created_cluster_master = False
+    for i, instance in enumerate(instances):
```

Comment:
Rather than "enumerate(instances)", can you use instances.keys()? I do not see where you use "instance" from enumerate(instances)

Reply:
yea, that was preexisting.  I didn't want to change too much.